File: backend/src/main/java/org/certificaciones/funproeibbackend/service/EvaluacionService.java
package org.certificaciones.funproeibbackend.service;

import org.certificaciones.funproeibbackend.dto.EvaluacionRequest;
import org.certificaciones.funproeibbackend.dto.EvaluacionResponse;

public interface EvaluacionService {
    EvaluacionResponse evaluar(EvaluacionRequest request);
    EvaluacionResponse obtenerPorPostulacion(Long idPostulacion);
}

File: frontend/src/app/core/services/evaluacion.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { EvaluacionRequest } from '../models/criterio.model';
import { EvaluacionResponse } from '../models/evaluacion.model';

@Injectable({ providedIn: 'root' })
export class EvaluacionService {
  constructor(private api: ApiService) {}

  evaluar(request: EvaluacionRequest): Observable<EvaluacionResponse> {
    return this.api.post<EvaluacionResponse>('/evaluaciones', request);
  }

  obtenerPorPostulacion(idPostulacion: number): Observable<EvaluacionResponse> {
    return this.api.get<EvaluacionResponse>(`/evaluaciones/postulacion/${idPostulacion}`);
  }
}
